core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanBuilder.scala

```diff
@@ -152,4 +156,84 @@ object SpanBuilder {
       }
     }
 
+  implicit final class SpanBuilderSyntax[F[_]](
+      private val builder: SpanBuilder[F]
+  ) extends AnyVal {
+
+    def translate[G[_]](fk: F ~> G, gk: G ~> F)(implicit
+        F: MonadCancel[F, _],
+        G: MonadCancel[G, _]
+    ): SpanBuilder[G] =
```

Comment:
> is the reason for this so that the primary API doesn't contain MonadCancel?

We need it in several places, for example: https://github.com/typelevel/otel4s/pull/264/files#diff-36bdd6d74392dca0ddc0d9c6312f06c778e77a49e87fbc5c6fc966c936171f1cR208

Reply:
> We need it in several places

oops, I worded that question really poorly. what's the reason it's an extension method and not in the trait?

Reply:
That way I do not need to deal with recursive definitions, e.g.:

```scala
trait Tracer[F[_]] { self =>
  def mapK[G[_]](fk: F ~> G): Tracer[G] = 
    new Tracer {
      ...
      def mapK[Q[_]](fk1: G ~> Q): Tracer[Q] = 
        self.mapK(fk.andThen(fk1)))
    }
}
```

I like your approach with a dedicated type `MappedK` way more.